frontend/src/components/user/UserManageModal.vue

```diff
@@ -0,0 +1,56 @@
+<script setup lang="ts">
+import { Button, Dialog } from '@/lib/primevue';
+
+import { RadioList } from '@/components/form';
+import { ROLES } from '@/utils/constants/application';
+
+// Emits
+const emit = defineEmits(['userManage:save']);
+
+// State
+const visible = defineModel<boolean>('visible');
+</script>
+
+<template>
+  <Dialog
+    v-model:visible="visible"
+    :draggable="false"
+    :modal="true"
+    class="app-info-dialog w-3"
+  >
+    <template #header>
+      <span class="p-dialog-title">Manage user role</span>
+    </template>
+    <div>Select role</div>
+    <RadioList
+      name="role"
+      :bold="false"
+      :options="ROLES"
+      class="mt-3 mb-4"
+    />
+    <div class="flex-auto">
+      <Button
+        class="mr-2"
+        label="Save"
+        type="submit"
+        icon="pi pi-check"
+        @click="
+          () => {
+            emit('userManage:save');
+            visible = false;
+          }
+        "
+      />
+      <Button
+        class="p-button-outlined mr-2"
+        label="Cancel"
+        icon="pi pi-times"
+        @click="
+          () => {
+            visible = false;
+          }
+        "
```

Comment:
Can simplify this to `@click='visible = false'` unless there needs to be multiple statements to execute.